rust/lance/src/io/commit.rs

```diff
@@ -206,7 +206,7 @@ fn fix_schema(manifest: &mut Manifest) -> Result<()> {
     for fragment in manifest.fragments.iter() {
         for file in fragment.files.iter() {
             for field_id in file.fields.iter() {
-                if !seen_fields.insert(*field_id) {
+                if !seen_fields.insert(*field_id) && *field_id != -2 {
```

Comment:
Should this just be `>= 0`?  Also, if you reverse the checks you can avoid ever inserting `-2` into `seen_fields` (although maybe that breaks something else down the road?)